### PR TITLE
Disallow certain modifiers in record parameters

### PR DESCRIPTION
**Pull request by jcouv**
Here's the general section of the spec on method parameters, for context: https://github.com/dotnet/csharplang/blob/master/spec/classes.md#method-parameters (note it doesn't have `in` yet)

### CODE REVIEW

File: proposals/records.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ record_body
 Record types are reference types, similar to a class declaration. It is an error for a record to provide
 a `record_base` `argument_list` if the `record_declaration` does not contain a `parameter_list`.
 
+Record parameters cannot use `ref`, `out` or `this` modifiers.
```
Comment (333fred) [inline proposals/records.md:30]: > (note it doesn't have in yet)

Does that mean we're allowing `in` here, or that we're not allowing `in` here and we'll need to update this when the spec updates? In either case, I think that should be explicitly clarified.

Comment (AlekseyTs) [inline proposals/records.md:30]: Should this be moved to the Primary Constructor section?

Comment (jcouv) [reply proposals/records.md:30]: `in` is allowed. If it were disallowed, we'd simply list it here as disallowed (no reason not to). I'd rather manage multiple copies of the list of allowed parameter modifiers as it continues to evolve.

Comment (jcouv) [reply proposals/records.md:30]: meant "avoid managing" (ie. the opposite of what I wrote ;-))

Comment (333fred) [reply proposals/records.md:30]: While I agree in principle, until we have an actual updated master list this just leaves it ambiguous. At some point we'll have to take this and turn it into an actual spec, and at that point we're going to have ask "well, was it intended to be allowed, or was it just not completely spec'ed and we'll do whatever the implementation does?

Comment (jcouv) [reply proposals/records.md]: Okay

**Pull request merged**